src/components/LazyLog/index.jsx

```diff
@@ -525,10 +618,54 @@ export default class LazyLog extends Component {
     }
   };
 
-  handleFormatPart = () => {
-    const { isSearching, searchKeywords } = this.state;
+  handleFormatPart = lineNumber => {
+    const {
+      isSearching,
+      searchKeywords,
+      resultLines,
+      currentResultsPosition,
+    } = this.state;
+    const { searchLikeBrowser } = this.props;
 
     if (isSearching) {
+      // If browser-search has started and we're on the line
+      // that has the search term that is selected
+      if (
+        searchLikeBrowser &&
+        resultLines &&
+        currentResultsPosition !== undefined &&
+        resultLines[currentResultsPosition] === lineNumber
+      ) {
+        // This finds which word in the line should be the highlighted one.
+        // For example, if we should be highlighting the 2nd match on line 18,
+        // this would set locationInLine to 2.
+        let locationInLine = 0;
+
+        for (let i = 0; i < currentResultsPosition + 1; i += 1) {
+          if (resultLines[i] === lineNumber) {
+            locationInLine += 1;
+          }
+        }
+
+        return searchFormatPart({
+          searchKeywords,
+          formatPart: this.props.formatPart,
+          caseInsensitive: this.props.caseInsensitive,
+          replaceJsx: (text, key) => (
+            <span key={key} className={searchMatch}>
```

Comment:
I guess I kind of like keeping these function based because it feels like it makes it easier to change things down the road. Like if we want to make the specially highlighted one flash or change color on hover, or wrap it in a div instead of a span or add an icon for some reason, we can do that a bit easier by passing in the function for `reactStringReplace` to use directly rather than just than a class name that searchFormatPart needs to hardcode how to handle.